src/extlibs/hrvo/hrvo_agent.h

```diff
@@ -88,18 +95,24 @@ class HRVOAgent : public Agent
      */
     VelocityObstacle createVelocityObstacle(const Agent &other_agent) override;
 
+    /**
+     * Computes the preferred velocity of this agent.
+     * relies on
+     * pref_speed, velocity and position
+     *
+     * updates pref_velocity
+     */
+    void computePreferredVelocity(double time_step);
```

Comment:
Lets make all of the `time_step` values to be of type `Duration`. Same as `computeNewVelocity` and `updatePrimitive`